### DISCUSSION TITLE
How to refresh the whole session's data?

### DISCUSSION THREAD

**Discussion by tht-activeviam** [atoti/atoti#637] [Q&A] [answered]
### Use case
 The updated data(csv, database, etc) arrives every day(or every other day). User want to reload/refresh cubes with new data without restarting the application.

### Possible solution
- Create a REST endpoint to trigger data reload event. Probably by admin user to call the REST endpoint.
- call `table.drop()` in every table and load the data again

Is there a better way to do?

**Comment by Skyscimitar** [comment, marked as answer]
Hi,

If the new data overwrites the old data (i.e the keys are the same) then you don't need to drop the old data. However if you need to ensure all the old data is gone then yes you will have to call `table.drop()` in every table and load the new data.

An alternative to creating a REST endpoint would be to create a background thread which periodically checks if there is any new data and refreshes the session if it finds any (you can check out this [example](https://github.com/atoti/project-template/blob/3b5da593f4e531d091c64f151c8359a6feddd1e1/app/app.py#L21-L24) in the project template).

**Reply by tibdex** [reply]
And in all cases, don't forget to use [`start_transaction()`](https://docs.atoti.io/0.6.6/lib/atoti/atoti.session.html#atoti.Session.start_transaction) to batch the operations ;).

**Reply by tht-activeviam** [reply]
Thanks @Skyscimitar and @tibdex
